Upgrade guide 2.2: Thread resolved as first-class citizen (#1793)
liveblocks/liveblocks: docs/pages/platform/upgrading/2.2.mdx

---
meta:
  title: "Upgrading to 2.2"
  parentTitle: "Upgrading"
  description: "Guide to upgrade to Liveblocks version 2.2"
---

> We are making Thread "resolved" a first-class citizen property. You don't have

Reviewer: ```suggestion
We are making “resolved”  a first-class citizen property on [Thread](...). You don't have
```
My computers broken, and I'm on mobile. Its difficult to make changes, could you link to the thread component in react-ui api reference? Also replace all straight quotes on the page with curly ones (e.g. '  to ’). Thanks!

Author: Done!

to use thread's metadata to set a thread as resolved anymore.

## All breaking changes are for Comments and only if you use a metadata "resolved"

> If you are using Comments without using a metadata `resolved`, there are no

Reviewer: ```suggestion
If you are using Comments without using the `resolved` metadata property, there are no
```

Reviewer 2: Good suggestion, @CTNicholas. There are a couple more instances of this.

Author: Fixed!

breaking changes for you! However, if you are using a thread's metadata
"resolved", and have logic based on the webhook event `threadMetadataUpdated`,
keep reading.

### Webhook events

When you upgrade to 2.2, our React component will update the thread first-class

> citizen property instead of the metadata. When a thread is marked as resolved or

Reviewer 2: This sentence is flowing a bit weirdly. Did you mean to say "update to" here?

Author: The default behaviours of the react component when clicking on "mark as resolved" will be to update the new property instead of the metadata.

unresolved, we will send the events `threadMarkedAsResolved` and

> `threadMarkedAsUnresolved` instead of `threadMetadataUpdated`.

Reviewer: Could you add links to all the events? Or where the link will be, if its not added yet

Author: Done

> Before upgrading to 2.2, you should update you webhook endpoint to process those

Reviewer 2: ```suggestion
Before upgrading to 2.2, you should update your webhook endpoint to process those
```

new events accordingly.

### REST API update thread metadata endpoint

If you use the REST endpoint

> [`update thread metadata`](/docs/api-reference/rest-api-endpoints#post-rooms-roomId-threads-threadId-metadata)

Reviewer 2: We call it "edit". Also, let's uppercase these REST API endpoints when referring to them?

```suggestion
[`Edit thread metadata`](/docs/api-reference/rest-api-endpoints#post-rooms-roomId-threads-threadId-metadata)
```

to update the metadata `resolved`, you should instead use the endpoints

> [`mark thread as resolved`](/docs/api-reference/rest-api-endpoints#post-rooms-roomId-threads-threadId-mark-as-resolved)

Reviewer 2: ```suggestion
[`Mark thread as resolved`](/docs/api-reference/rest-api-endpoints#post-rooms-roomId-threads-threadId-mark-as-resolved)
```

and

> [`mark thread as unresolved`](/docs/api-reference/rest-api-endpoints#post-rooms-roomId-threads-threadId-mark-as-unresolved).

Reviewer 2: ```suggestion
[`Mark thread as unresolved`](/docs/api-reference/rest-api-endpoints#post-rooms-roomId-threads-threadId-mark-as-unresolved).
```

Note that updating the metadata "resolved" automatically updates the first-class

> citizen property.

Reviewer 2: Will it also update/set the `metadata.resolved` property or not? (I forgot what we decided here in the end.)

Author: yes it also updates the metadata

### Migration plan

Here is the plan to migrate the metadata `resolved` to the first-class citizen
property.

1. Support the new events `threadMarkedAsResolved` and

>    `threadMarkedAsUnresolved` in your webhook endpoint.

Reviewer 2: Do you mean to enable them in your dashboard here? Maybe put a link to that place then, and maybe make this a bit clearer with a GIF, like we do [here](https://liveblocks.io/docs/platform/webhooks#creating-an-endpoint)?

Author: Yes, enabling the in your dashboard, but also adding the logic to process those events in your back-end.

> 2. Upgrade liveblocks packages to 2.2.

Reviewer 2: ```suggestion
2. Upgrade all Liveblocks packages to 2.2. Run `npx create-liveblocks-app@latest upgrade` to do so.
```

> 3. Update your back-end to use the new "mark thread as resolved/unresolved"

Reviewer 2: @CTNicholas Do we say back-end or backend?

Reviewer: Good catch! We actually say it the third valid way, "back end"

Author: Fixing it!
